scripts/research/lib/batch02_source_policy.py

```
@@ -0,0 +1,676 @@
+) -> list[Path]:
+    out: list[Path] = []
+    for path in research.rglob("*.py"):
+        try:
+            tree = ast.parse(
+                path.read_text(encoding="utf-8"),
+                filename=str(path),
+            )
+        except (OSError, UnicodeDecodeError, SyntaxError) as exc:
+            raise Batch02SourcePolicyError(
+                f"unable to inspect potential Batch02 source {path}: {exc}"
+            ) from exc
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)